package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -26,24 +26,59 @@ def strip_encoding(dataset_name: str) -> str:
 
     @property
     def layers_mapping(self):
-        """Return layer mapping: dataset_full_name -> layer it belongs to in the catalog"""
+        """Return layer mapping: dataset_full_name -> layer it belongs to in the catalog
+        From kedro-datasets 1.3.0 onwards, the 'layers' attribute is defined inside the 'metadata'
+        under 'kedro-viz' plugin.
+
+        Catalog before kedro-datasets 1.3.0:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            layers: raw
+
+        Catalog from kedro-datasets 1.3.0 onwards:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            metadata:
+                kedro-viz:
+                    layers: raw
+
+        Currently, Kedro up to 18.x and kedro-viz up to 6.x support both formats =.
+        However, support for the old format will be discontinued from Kedro 19.x and kedro-viz
+        7.x onwards.
+        It's recommended to follow the newest format for defining layers in the catalog.
+        """
         if self._layers_mapping is not None:
             return self._layers_mapping
 
+        self._layers_mapping = {}
+
+        # TODO - remove the old method of mapping layers when we upgrade to viz 7.x
+        # Maps layers according to the old format
         if self._catalog.layers is None:
             self._layers_mapping = {
                 self.strip_encoding(dataset_name): None
                 for dataset_name in self._catalog._data_sets
             }
         else:
-            self._layers_mapping = {}
             for layer, dataset_names in self._catalog.layers.items():
                 self._layers_mapping.update(
                     {
                         self.strip_encoding(dataset_name): layer
                         for dataset_name in dataset_names
                     }
                 )
+
+        # Maps layers according to the new format
+        for dataset_name in self._catalog._data_sets:
+            dataset = self._catalog._get_dataset(dataset_name)
+            metadata = getattr(dataset, "metadata", {})
+            if not metadata:
+                continue
+            dataset_layer = metadata.get("kedro-viz", {}).get("layer")
```

Comment:
I'd definitely go for [EAFP](https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python) for the clearest way to write a nested dictionary lookup. This way you avoid many `get`s and `continue` e.g. https://stackoverflow.com/questions/43491287/elegant-way-to-check-if-a-nested-key-exists-in-a-dict
```
try:
    dataset_layer = dataset.metadata["kedro-viz"]["layer"]
except AttributeError, KeyError:
    pass
else:
    # update self._layers_mapping
```